.github/workflows/release-manual.yaml

```diff
@@ -0,0 +1,88 @@
+# This workflow only build and upload to CF R2 for a specific contract name passed as input.
+name: Manual Release (Only build and upload to CF R2)
+
+on:
+  workflow_dispatch:
+    inputs:
+      contract-path:
+        description: "Path of the contract to release (e.g. contracts)"
+        required: true
+        type: string
+      contract-name:
+        description: "Name of the contract to release (e.g. stellar-axelar-gateway)"
+        required: true
+        type: string
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  prepare-release:
+    name: Prepare release metadata
+
+    runs-on: blacksmith-4vcpu-ubuntu-2204
+
+    outputs:
+      releases: ${{ steps.prepare-matrix.outputs.releases }}
+      commit_hash: ${{ steps.get-commit-hash.outputs.hash }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.PAT_TOKEN }}
```

Comment:
to allow arbitrary overwrite, I'd prefer if we take the `version` as input and checkout the contract at that tag. This ensures that we're building from the right tag, rather than from main which is a different version. If the contract needs a patch, then a new release should be created anyways.